sdk/python/kfp_tekton/compiler/_op_to_template.py

```diff
@@ -319,7 +319,32 @@ def _process_output_artifacts(outputs_dict: Dict[Text, Any],
         mounted_artifact_paths = []
         for artifact in outputs_dict['artifacts']:
             artifact_name = artifact_to_result_mapping.get(artifact['name'], artifact['name'])
-            if artifact['name'] in replaced_param_list:
+            if artifact['path'] == 'logs_to_S3':
+                copy_artifacts_step['script'] = copy_artifacts_step['script'] + \
+                    'mc cp /var/log/containers/$TASKRUN*$NAMESPACE*.log' + \
+                    ' storage/%s/artifacts/$PIPELINERUN/$PIPELINETASK/\n' % (bucket)
+                copy_artifacts_step['env'].append({
+                    'name': 'TASKRUN',
+                    'valueFrom': {
+                        'fieldRef': {
+                            'fieldPath': "metadata.labels['tekton.dev/taskRun']"
+                        }
+                    }
+                })
+                volume_mount_step_template.extend([{
+                    'name': 'varlog',
+                    'mountPath': '/var/log'
+                }, {
+                    'name': 'varlibdockercontainers',
+                    'mountPath': '/var/lib/docker/containers',
+                    'readOnly': True
+                }])
+                volume_template.extend([{
+                    'name': 'varlog', 'hostPath': {'path': '/var/log'}
+                }, {
+                    'name': 'varlibdockercontainers', 'hostPath': {'path': '/var/lib/docker/containers'}
+                }])
+            elif artifact['name'] in replaced_param_list:
```

Comment:
It should be `if` not `elif`, right now you are removing all the other artifacts when logging is enabled.

Reply:
@Tomcli , Thanks for your comments, It should be elif.
because it's under the loop of `for artifact in outputs_dict['artifacts']:`, if it's logging artifact, we should prevent it to continue others.

Reply:
Thanks, probably some logic modified the artifact list. Let's try to find out where's the real bug.

Reply:
Oh sorry I think I was running the test with artifacts disabled. It's working now after I turn it back on. You can resolve this comment now.